### PR TITLE
Keep individual segment's updates in a single Etcd operation

### PR DESCRIPTION
**Pull request by soothing-rain**
/kind improvement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

File: internal/metastore/kv/datacoord/kv_catalog.go [hunk 1]
```diff
@@ -102,23 +104,37 @@ func (kc *Catalog) AlterSegments(ctx context.Context, newSegments []*datapb.Segm
...
+			if err := kc.Txn.MultiSave(kvsPiece); err != nil {
+				log.Error("failed to call MultiSave", zap.Error(err))
+				return err
+			}
+			kvsPiece = make(map[string]string)
+			currSize = 0
+		}
+		maps.Copy(kvsPiece, kvs)
+		currSize += len(kvs)
 	}
```
Comment (congqixia) [inline internal/metastore/kv/datacoord/kv_catalog.go:131]: This piece of logic is implemented inside `etcd.SaveByBatch`

Comment (soothing-rain) [reply internal/metastore/kv/datacoord/kv_catalog.go:131]: Yes but not by segment, so there's a chance that kvs of a same segment are split into different operations.

Comment (congqixia) [reply internal/metastore/kv/datacoord/kv_catalog.go:131]: If the transaction number of one segment is greater than 64, it still needs to be split into multiple operation.

Comment (soothing-rain) [reply internal/metastore/kv/datacoord/kv_catalog.go:131]: In this case we won't have more than 64 kvs (ops) for a single segment.

Comment (congqixia) [reply internal/metastore/kv/datacoord/kv_catalog.go:131]: If collection has more than  32 fields, it shall have 32*2+1 ops for save one segment

Comment (soothing-rain) [reply internal/metastore/kv/datacoord/kv_catalog.go]: Ah I see what you mean, that's true. If that case these binlog kvs could still get separated into multiple OPs.

**Comment by soothing-rain** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**